mtags/src/main/scala-3/scala/meta/internal/pc/printer/ShortenedNames.scala

```diff
@@ -125,20 +132,24 @@ class ShortenedNames(
               sym: Symbol,
               prev: List[Symbol],
               ownersLeft: List[Symbol],
-          ): Type =
+          ): WithRenames[Type] =
             ownersLeft match
               case Nil =>
                 val short = ShortName(sym)
-                if tryShortenName(short) then TypeRef(NoPrefix, sym)
-                else TypeRef(loop(prefix, Some(short)), sym)
+                if tryShortenName(short) then
+                  WithRenames(TypeRef(NoPrefix, sym))
+                else loop(prefix, Some(short)).map(TypeRef(_, sym))
               case h :: tl =>
                 indexedContext.rename(h) match
                   // case where the completing symbol is renamed in the context
                   // for example, we have `import java.lang.{Boolean => JBoolean}` and
                   // complete `java.lang.Boolean`. See `CompletionOverrideSuite`'s `rename'.
                   case Some(rename) =>
-                    PrettyType(
-                      (rename :: prev.map(_.name)).mkString(".")
+                    WithRenames(
+                      PrettyType(
+                        (rename :: prev.map(_.name)).mkString(".")
+                      ),
+                      Map(rename -> h.showName),
```

Comment:
Wouldn't it cause less changes to mark in the rename map, what was renamed? To avoid too many changes?

Reply:
Yep, changed that.